Testio APP

mscmnc opened the pull request:
Hello, It was the best party I ever had :). It's my first React app.

ArnasJuskevicius commented:
Hello, Aurimas,

first of all, congratulations on completing the assignment. I hope you had fun :).

# Overall

Could you write your thoughts on the task to the description of the PR or the Readme file:

- What would you do differently?
- What were the challenges?
- How to test/dev launch/production build the app and etc.

## Observations and Tips

### Common
- Why is all of the code contained in one commit?

- [js] Code style is inconsistent. Some files contains 2 space other 4. I recommend check out eslint and prettier.

- [scss] It's a good practice to save color values to variable. https://sass-lang.com/guide

# Login page
## Tips

- [js] LocalStorage API is used in a few different places, with hard coded values. This could be change to variable

- [js] LocalStorage API key is in Lithuanian would, would change to English.

- [js] Getting input values could be improved by using react way. https://reactjs.org/docs/forms.html

- [js] Input component could be updated by using destructuring. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment

# Servers page
## Tips

- [js] Lithuanian names should be avoided. Image if you would work with international team. :)

- [js] After successful call, data is appended to DOM. This should be avoided while using react. Better solution, is to save to state this could be component state or some other like redux, mobx.

mscmnc commented:
> Hello, Aurimas,
> 
> first of all, congratulations on completing the assignment. I hope you had fun :).
> 
> # Overall
> Could you write your thoughts on the task to the description of the PR or the Readme file:
> 
> * What would you do differently?
**_I would add posibility to sort results by Server and Distance. It will be a small arrow „↓“ button next to each criteria._**
> * What were the challenges?
**_With HTML5, CSS3 and SASS parts I was already familiar so it wasn‘t very complicated. In React part almost everything was challenging because I am a beginner :). As I mentioned before, it was my first React app._**
> * How to test/dev launch/production build the app and etc.
**_To run development mode you need to write npm install and then npm start. There is no possibility to run my tests, I didn‘t do that part, but if you want to try you can run one default test with command npm test. For build production mode use command npm run-script build._** 
> 
> ## Observations and Tips
> ### Common
> * Why is all of the code contained in one commit?
**_I thought that  I will commit everything when all will be finished._**
> * [js] Code style is inconsistent. Some files contains 2 space other 4. I recommend check out eslint and prettier.
 **_I installed npm package „prettier-eslint“ and added the script to fix the code._**
> * [scss] It's a good practice to save color values to variable. https://sass-lang.com/guide
**_Yes, I see that. I did this only for a few color values, I missed others. Everything is fixed now! :)._**  
> 
> # Login page
> ## Tips
> * [js] LocalStorage API is used in a few different places, with hard coded values. This could be change to variable
**_I can‘t find these few places :). Maybe I am missing something? In login page I use localStorage with hard-coded value „token“ only one time and one time in other react component LoggedPage?_**
> * [js] LocalStorage API key is in Lithuanian would, would change to English.
**_Fixed._**
> * [js] Getting input values could be improved by using react way. https://reactjs.org/docs/forms.html
**_Fixed._**
> * [js] Input component could be updated by using destructuring. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment
 **_I hope I did it right._**
> 
> # Servers page
> ## Tips
> * [js] Lithuanian names should be avoided. Image if you would work with international team. :)
**_Fixed :)_**
> * [js] After successful call, data is appended to DOM. This should be avoided while using react. Better solution, is to save to state this could be component state or some other like redux, mobx.
**_Fixed._**

mscmnc commented:
Hi, Arnas,
I finished my course about ReactJS and decided to improve my skills with your task again. 
It would be awesome if you could find time to check what I have done and give me some comments or advices. Thanks in advance  :)
Short list of what I have done:
•	Added async Redux.
•	Added axios.
•	Added authentication.
•	Added guard, you can‘t visit /logged page if you are not authenticated.
•	Added error handlers.
•	Added spinner.
•	Added couple simple tests with Jest and enzyme.
Login Page: 
•	I made a structure for login form. 
•	If there is no entered information in input field, button is disabled.
•	Fixed input width.
•	Fixed input icon.
Logged Page:
•	Added components.
•	Fixed scroll bar, now you can scroll only on results.

Also I created a new project with ReactJS. It‘s image finder, API from Unsplash.
Code: https://github.com/mscmnc/webapp Live demo: https://mscmnc.github.io/webapp/

Best wishes and happy new year! :).
